docs/install/Knative-with-any-k8s.md

```diff
@@ -1,89 +1,474 @@
+   kubectl --namespace kourier-system get service kourier
    ```
 
+   Save this for configuring DNS below.
+
+{{< /tab >}}
+{{< /tabs >}}
+
+1. Configure DNS
+
+   <!-- This indentation is important for things to render properly. -->
+   {{< tabs name="serving_dns" >}}
+{{% tab name="Magic DNS (xip.io)" %}}
+We ship a simple Kubernetes Job called "default domain" that will (see caveats) configure Knative Serving to use <a href="http://xip.io">xip.io</a> as the default DNS suffix.
+
+```bash
+kubectl apply --filename https://github.com/knative/serving/releases/download/{{< version >}}/serving-default-domain.yaml
+```
+
+**Caveat**: This will only work if the cluster LoadBalancer service exposes an IPv4 address, so it will not work with IPv6 clusters, AWS, or local setups like Minikube.  For these, see "Real DNS".
+{{< /tab >}}
+
+
+{{% tab name="Real DNS" %}}
+To configure DNS for Knative, take the External IP or CNAME from setting up networking, and configure it with your DNS provider as follows:
+
+- If the networking layer produced an External IP address, then configure a wildcard A record for the domain:
+
+    ```
+    # Here knative.example.com is the domain suffix for your cluster
+    *.knative.example.com == A 35.233.41.212
+    ```
+
+- If the networking layer produced a CNAME, then configure a CNAME record for the domain:
+
+    ```
+    # Here knative.example.com is the domain suffix for your cluster
+    *.knative.example.com == CNAME a317a278525d111e89f272a164fd35fb-1510370581.eu-central-1.elb.amazonaws.com
+    ```
+
+Once your DNS provider has been configured, direct Knative to use that domain with:
+
+```bash
+# Replace knative.example.com with your domain suffix
+kubectl patch configmap/config-domain \
+  --namespace knative-serving \
+  --type merge \
+  --patch '{"data":{"knative.example.com":""}}'
+```
+
+{{< /tab >}}
+{{< /tabs >}}
+
+1. Monitor the Knative components until all of the components show a `STATUS` of `Running` or `Completed`:
+
+   ```bash
+   kubectl get pods --namespace knative-serving
+   ```
+
+At this point, you have a basic installation of Knative Serving!
+
+
+### Optional Serving extensions
+
+{{< tabs name="serving_extensions" >}}
+{{% tab name="HPA autoscaling" %}}
+
+Knative also supports the use of the Kubernetes Horizontal Pod Autoscaler (HPA)
+for driving autoscaling decisions.  The following command will install the
+components needed to support HPA-class autoscaling:
+
+```bash
+kubectl apply --filename https://github.com/knative/serving/releases/download/{{< version >}}/serving-hpa.yaml
+```
+
+<!-- TODO(https://github.com/knative/docs/issues/2152): Link to a more in-depth guide on HPA-class autoscaling -->
+
+{{< /tab >}}
+
+{{% tab name="TLS with cert-manager" %}}
+
+Knative supports automatically provisioning TLS certificates via cert-manager.  The following commands will install the components needed to support the provisioning of TLS certificates via cert-manager:
```

Comment:
```suggestion
Knative supports automatically provisioning TLS certificates via [cert-manager](https://cert-manager.io/docs/).  The following commands will install the components needed to support the provisioning of TLS certificates via cert-manager:
```

Reply:
Please check to make sure that's the right cert-manager :)

Reply:
it is!